finetune/encoding/target_encoders.py

```diff
@@ -77,6 +78,28 @@ def inverse_transform(self, one_hot):
                     break
         return ys
 
+class NoisyLabelEncoder(LabelEncoder, BaseEncoder):
+
+    # Overriding the fit method...
+    # Fit method may not be necessary at all if pandas is
+    # consistent about how it chooses columns
+    # TODO: Check
+    def fit(self, y):
+        self.classes_ = list(pd.DataFrame(y[:1]).columns)
```

Comment:
Assumption: y[0] contains all class labels

Reply:
Yeah that is an assumption. I was prematurely optimizing--didn't want to do the whole pd.DataFrame conversion just to get labels. Now that I think about it, normal encoders do need to read through all the labels to fit anyway... Haven't profiled this yet though so all speculation.